lightning/src/ln/onion_message.rs

```diff
@@ -154,6 +163,9 @@ impl<Signer: Sign, K: Deref> OnionMessager<Signer, K>
 
 impl OnionMessageHandler for OnionMessager {
 	fn handle_onion_message(&self, peer_node_id: &PublicKey, msg: &msgs::OnionMessage) {
+		// calls:
+		// * onion_utils::decode_next_message_hop
+		// * onion_utils::next_hop_packet_pubkey
 	}
```

Comment:
I take it users would typically just implement `InvoiceHandler` but we're giving the option to implement `OnionMessageHandler` to handle custom onion messages? Should we have a separate hook for that?

Also, is there a case where users would need custom logic outside of handling custom messages? Just wondering if it makes sense for users to ever do the decoding / forwarding. Maybe this akin to `ChannelMessageHandler` and `ChannelManager`, where the latter is almost always used.

Reply:
> Maybe this akin to ChannelMessageHandler and ChannelManager, where the latter is almost always used.

I think this is the case. I don't think users would want to do the decoding themselves, seems likely they'd only want to implement handling of custom onion messages if anything